### PR TITLE
feat: enable extending user tsconfig

### PR DESCRIPTION
**Pull request by robBowes**
Closes https://github.com/sveltejs/kit/issues/6868 by allowing the user to specify a root tsconfig to extend. This is useful for monorepos where the is a base tsconfig file at the root that needs to be extended.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by Rich-Harris** [comment]
Thank you. `extends` probably isn't the right name — it should be something tsconfig-specific.

I wonder if a more future-proof approach might be to allow the generated config to be augmented in arbitrary ways? Something like this:

```js
// svelte.config.js
import path from 'path';

export const config = {
  kit: {
    tsconfig: (tsconfig) => {
      tsconfig.extends = path.resolve('../tsconfig.json');
    }
  }
};
```

Or this:

```js
// svelte.config.js
import path from 'path';

export const config = {
  kit: {
    tsconfig: (tsconfig) => {
      return {
        ...tsconfig,
        extends: path.resolve('../tsconfig.json')
      };
    }
  }
};
```

I don't know if `tsconfig` is the best name for the option.

**Comment by dummdidumm** [comment]
I like the second version of Rich's proposal. I think `tsconfig` is the right name.

File: .changeset/warm-pumas-pull.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+Allow sveltekit tsconfig to be extended by adding extends option to svelte config
```
Comment (benmccann) [inline .changeset/warm-pumas-pull.md:5]:
```suggestion
'@sveltejs/kit': minor
---

feat: allow SvelteKit tsconfig to be extended by adding extends option to svelte config
```

**Comment by Rich-Harris** [comment]
going to close this in favour of #8606 which implements the idea above — thanks!

**Pull request closed without merging**